File: services/vets-service/src/test/java/com/github/antrakos/petclinic/vets/config/TestContainersConfig.java
package com.github.antrakos.petclinic.vets.config;

import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
import org.springframework.context.annotation.Bean;
import org.testcontainers.containers.MongoDBContainer;

@TestConfiguration(proxyBeanMethods = false)
public class TestContainersConfig {
    @Bean
    @ServiceConnection
    public MongoDBContainer mongoDBContainer() {
        return new MongoDBContainer("mongo:6.0").withReuse(true);
    }
}

File: services/visits-service/src/test/kotlin/com/github/antrakos/petclinic/visits/config/TestContainersConfig.kt
package com.github.antrakos.petclinic.visits.config

import org.springframework.boot.test.context.TestConfiguration
import org.springframework.boot.testcontainers.service.connection.ServiceConnection
import org.springframework.context.annotation.Bean
import org.testcontainers.containers.PostgreSQLContainer

@TestConfiguration(proxyBeanMethods = false)
class TestContainersConfig {
    @Bean
    @ServiceConnection
    fun postgreSQLContainer() = PostgreSQLContainer("postgres:15")
        .withReuse(true)
}
